plugins/utils/lookup_apps.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""
+Lookup the "apps" directory to generate apps configuration
+"""
+import os
+import re
+
+from ansible.plugins.filter.core import to_nice_yaml
+
+PROJECT_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "../.."))
+APPS_DIRECTORY = os.path.join(PROJECT_ROOT, "apps")
+IGNORE_DIRS_REGEX = "^{}/_.*(/.*)*$".format(APPS_DIRECTORY)
+APP_VOLUMES_DIR = "_volumes"
+SERVICE_CONFIG_DIR = "_configs"
+
+
+def lookup_apps(apps_path):
+    """
+    Lookup ready-to-deploy applications following recommended Arnold applications'
+    tree, e.g. for the "richie" application and the "apps" ``apps_path``:
+
+        apps/richie
+        ├── templates
+        │   ├── elasticsearch
+        │   │   ├── dc.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── nginx
+        │   │   ├── _configs
+        │   │   │   └── richie.conf.j2
+        │   │   ├── dc.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── postgresql
+        │   │   ├── dc.yml.j2
+        │   │   ├── ep.yml.j2
+        │   │   └── svc.yml.j2
+        │   ├── richie
+        │   │   ├── dc.yml.j2
+        │   │   ├── job_collectstatic.yml.j2
+        │   │   ├── job_db_migrate.yml.j2
+        │   │   ├── job_regenerate_indexes.yml.j2
+        │   │   ├── route.yml.j2
+        │   │   └── svc.yml.j2
+        │   └── _volumes
+        │       ├── media.yml.j2
+        │       └── static.yml.j2
+        └── vars
+            └── main.yml
+
+    Analysing this tree with ``lookup_apps`` should return the following data structure:
+
+        [
+            {
+                "name": "richie",
+                "services": [
+                    {
+                        "configs": [
```

Comment:
In my head the template are not in this var but find by the playbooks with the app.name and service.name.
If we defined the apps var we don't need to liste the template on the var.
For me the data structure must be :+1: 
```
 [
            {
                "name": "richie",
                "services": [
                    {
                        "name": "nginx"
                    },
                    {
                        "name": "postgresql"
                    },
                    {
                        "name": "elasticsearch"
                    },
                    {
                        "name": "richie"
                    }
                ]
            }
        ]
```

Reply:
We've discussed this. I'll make a new PR to improve apps configuration overriding mechanism.